### PR TITLE
Fix get photos bug with many2many

### PR DESCRIPTION
**Pull request by carrosa**
Also made a new PhotoTagReferenceDto to help handle this better.

**Before testing:**
I have made some changes in the Photo upload function, so now PhotoTagReferences are also stored in DB when storing a new photo.
To avoid errors you should run the script ./scripts/droprecreatedb.sh before restarting maven, and then seed the database with get request to localhost:8080/seed

**To test:**
Test that you do not get "NoSuchMethodError" when trying to get photos from the following endpoints:

/photos
/photos/motive/{motiveId}
/photos/carousel

### CODE REVIEW

File: src/main/kotlin/no/fg/hilflingbackend/controller/PhotoGangBangerController.kt [hunk 1]
```diff
@@ -43,9 +41,24 @@ class PhotoGangBangerController {
...
-    repository.create(photoGangBangerDto)
+    repository.create(dto)
+
+  @PostMapping
+  fun create2(
+    @RequestBody dto: SemesterStart
+  ): Int {
+    return 1
+  }
+
```
Comment (NikZy) [inline src/main/kotlin/no/fg/hilflingbackend/controller/PhotoGangBangerController.kt:56]: Fjerne dette? :)

File: src/main/kotlin/no/fg/hilflingbackend/service/PhotoService.kt [hunk 2]
```diff
@@ -416,4 +417,8 @@ class PhotoService(
     return photoRepository
       .findAll()
   }
+
+  fun test(): Any {
+    return photoRepository.findAll()
+  }
```
Comment (NikZy) [inline src/main/kotlin/no/fg/hilflingbackend/service/PhotoService.kt:423]: Fjerne? :)

**Review by NikZy — APPROVED** [review]

File: src/main/kotlin/no/fg/hilflingbackend/controller/PhotoGangBangerController.kt (hunk 1 above)
Comment (carrosa) [reply src/main/kotlin/no/fg/hilflingbackend/controller/PhotoGangBangerController.kt:54]: Lar stå fordi dette er fra PRen jeg egentlig skulle jobbe med. Blir fjernet ila. helgen :)))

File: src/main/kotlin/no/fg/hilflingbackend/service/PhotoService.kt (hunk 2 above)
Comment (carrosa) [reply src/main/kotlin/no/fg/hilflingbackend/service/PhotoService.kt:422]: Jepp! ;)

**Review by taheeraahmed — APPROVED** [review]
Noice!!

**Pull request merged**